### PR TITLE
[feture](Nereids) support delete from without using

### PR DESCRIPTION
**Pull request by morrySnow**
## Proposed changes

Issue Number: close #xxx

support sql:
```sql
DELETE FROM <table_name> WHERE <predicates>
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by englefly — APPROVED** [review]

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 1]
```diff
@@ -62,9 +62,9 @@ statement
         SET updateAssignmentSeq
         fromClause?
         whereClause                                                    #update
-    | explain? cte? DELETE FROM tableName=multipartIdentifier tableAlias
-        (PARTITION partition=identifierList)?
-        (USING relation (COMMA relation)*)
+    | explain? cte? DELETE FROM tableName=multipartIdentifier
+        partitionSpec? tableAlias
+        (USING relation (COMMA relation)*)?
```
Comment (xiacongling) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:67]: why do we change the order of `tableAlias` and `partitionSpec`? the modified syntax does not follow the documentation and introduces compatibility issues.

**Pull request merged**